q2_demux/_subsample.py

```diff
@@ -0,0 +1,95 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2018, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os
+import gzip
+import random
+
+import pandas as pd
+
+from q2_types.per_sample_sequences import (
+    SingleLanePerSampleSingleEndFastqDirFmt,
+    SingleLanePerSamplePairedEndFastqDirFmt,
+    FastqManifestFormat)
+
+from q2_demux._demux import _read_fastq_seqs, _write_metadata_yaml
+
+
+def subsample_single(sequences: SingleLanePerSampleSingleEndFastqDirFmt,
+                     fraction: float
+                     ) -> SingleLanePerSampleSingleEndFastqDirFmt:
+    result = SingleLanePerSampleSingleEndFastqDirFmt()
```

Comment:
This is a nifty trick I learned from @ebolyen when working on q2-cutadapt. Since `SingleLanePerSampleSingleEndFastqDirFmt` is a subclass of `CasavaOneEightSingleLanePerSampleDirFmt`, the filename requirements are the same. If we instantiate a `CasavaOneEightSingleLanePerSampleDirFmt`, we can create the new dir fmt without worrying about the manifest --- we can just let the transformer do the work for us! 🥜 
```suggestion
    result = CasavaOneEightSingleLanePerSampleDirFmt()
```